packages/react-performance/README.md

```diff
@@ -0,0 +1,261 @@
+
+function App() {
+  return (
+    <PerformanceContext.Provider value={performance}>
+      {/* The rest of your app */}
+    </PerformanceContext.Provider>
+  );
+}
+```
+
+#### Display data using NavigationListener
+
+Now that we have access to `PerformanceContext` we can use the other components and hooks offered by `@shopify/react-performance` anywhere in our tree. To demonstrate, we'll create a component called `LastNavigationDetails` and use it to display some basic data about our app's performance.
+
+```tsx
+// LastNavigationDetails.tsx
+import React, {useState} from 'react';
+import {useNavigationListener, Navigation} from '@shopify/react-performance';
+
+export function LastNavigationDetails() {
+  const [lastNavigation, setLastNavigation] = useState<Navigation | null>(null);
+
+  useNavigationListener(navigation => {
+    setLastNavigation(navigation);
+  });
+
+  if (lastNavigation == null) {
+    return <p>no data</p>;
+  }
+
+  const {duration, isFullPageNavigation, target};
+  const navigationType = isFullPageNavigation
+    ? 'full page navigation'
+    : 'single-page-app style navigation';
+  return (
+    <p>
+      The last navigation was to {target}. It was a {navigationType} navigation
+      which took {duration / 1000} seconds to complete.
+    </p>
+  );
+}
+```
+
+We can render this component anywhere in our tree, but lets do so in our App component from earlier.
+
+```tsx
+// App.tsx
+import React from 'react';
+import {Performance, PerformanceContext} from '@shopify/react-performance';
+import {LastNavigationDetails} from './LastNavigationDetails';
+
+// in a Server-Side-Rendering enabled app you will likely only want to instantiate this if `document` is defined.
+const performance = new Performance();
+
+function App() {
+  return (
+    <PerformanceContext.Provider value={performance}>
+      {/* The rest of your app */}
+      <LastNavigationDetails />
+    </PerformanceContext.Provider>
+  );
+}
+```
+
+Now our developers will have access to up-to-date information about how long initial page-loads and incremental navigations take to complete without needing to open their devtools or dig into DOM APIs manually.
+
+### Reporting data to the server
+
+Performance metrics data is most useful when it can be aggregated and used to make smart optimization decisions or warn you when your app is beginning to become slow. To aggregate such data you'll need to be able to send it to a server. `@shopify/react-performance` provides the `PerformanceReport` component to facilitate this.
+
+> This section is a work in progress. It will be filled out as helper libraries for marshalling data on the server are added to Quilt.
+
+## API
+
+Most APIs provided by this package are available as both hooks (ie. `useNavigationListener`), as well as components (ie. `<NavigationListener />`).
+
+This package also re-exports all of the API of [`@shopify/performance`](https://www.npmjs.com/package/@shopify/performance).
+
+### Hooks
+
```

Comment:
Maybe it will be good to add some explanation on what life cycle events are before starting.

We can copy those from web's dev UI.

<img width="315" alt="Screen Shot 2019-10-02 at 4 14 44 PM" src="https://user-images.githubusercontent.com/1610169/66078432-02d1b700-e530-11e9-9ce4-98acb9f16e97.png">

Reply:
I avoided spelling out some of the details of what `@shopify/react-performance`'s events are since ultimately this library is just a react-based abstraction on top of that, but looking at that library it seems like it's docs are pretty lacking. Probably we'll want to make those a lot better in a subsequent PR and then add more link-outs to those docs from this one.